### PR TITLE
feat: Added support for Code Signing Configuration

### PR DESCRIPTION
**Pull request by tsoe77**
## Description

This PR adds code_signing_config_arn support.

The change doesn't effect any other things except it takes in code_signing_config_arn if user wants to provide one.

### CODE REVIEW

**Comment by tsoe77** [comment]
@bryantbiggs I appreciate your review and feedback for #262. I addressed all of those in this PR. Could I get your review for this please?

**Comment by agjmills** [comment]
👍 looks good to me 

@bryantbiggs thoughts?

**Review by bryantbiggs — APPROVED** [review]
Looks good to me, @antonbabenko what do you think

**Comment by antonbabenko** [comment]
Thanks to everyone! I will review it and merge it later this week. I have some other things to do before.

File: examples/code-signing/README.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
+# AWS Lambda Code Signing example
```
Comment (antonbabenko) [inline examples/code-signing/README.md:1]: Please update "examples" sections in the main README

**Review by antonbabenko — CHANGES_REQUESTED** [review]
I tried to run example/code-signing and this was the error:
```
╷
│ Error: error starting Signing Job: ValidationException: StartSigningJobRequest:source:version should not be empty
│ {
│   RespMetadata: {
│     StatusCode: 400,
│     RequestID: "35c2c96f-8ac7-488d-a2a2-779a9aad584a"
│   },
│   Code_: "InvalidJobParameters",
│   Message_: "StartSigningJobRequest:source:version should not be empty"
│ }
│ 
│   with aws_signer_signing_job.this,
│   on main.tf line 52, in resource "aws_signer_signing_job" "this":
│   52: resource "aws_signer_signing_job" "this" {
│ 
╵
```

Please take a look and fix it.

**Comment by tsoe77** [comment]
> I tried to run example/code-signing and this was the error:
> 
> ```
> ╷
> │ Error: error starting Signing Job: ValidationException: StartSigningJobRequest:source:version should not be empty
> │ {
> │   RespMetadata: {
> │     StatusCode: 400,
> │     RequestID: "35c2c96f-8ac7-488d-a2a2-779a9aad584a"
> │   },
> │   Code_: "InvalidJobParameters",
> │   Message_: "StartSigningJobRequest:source:version should not be empty"
> │ }
> │ 
> │   with aws_signer_signing_job.this,
> │   on main.tf line 52, in resource "aws_signer_signing_job" "this":
> │   52: resource "aws_signer_signing_job" "this" {
> │ 
> ╵
> ```
> 
> Please take a look and fix it.

This error occurs if S3 isn't ready to do the versioning yet. S3 object has version id of `null` when versioning is still propagating for the first time. I can reproduce the error as well but if I re-run the same in 5-10mins, it can pick up the version within the module itself. I will put a sleep 15mins in there.  

> **Note**
> 
> If you enable versioning on a bucket for the first time, it might take a short amount of time for the change to be fully > propagated. We recommend that you wait for 15 minutes after enabling versioning before issuing write operations (PUT or DELETE) on objects in the bucket. 
> [Enabling versioning on buckets](https://docs.aws.amazon.com/AmazonS3/latest/userguide/manage-versioning-examples.html)

**Comment by tsoe77** [comment]
@antonbabenko I added 5mins timer before uploading the s3 object in the example, example will run fine at first attempt now. Example README is also updated with time_sleep resource information and main README's example section has been updated as well.

**Review by antonbabenko — APPROVED** [review]
Looks good. I have updated the example a bit and verified it.

**Comment by antonbabenko** [comment]
This PR is included in version 4.2.0 :tada:

**Pull request merged**